src/Storages/StorageDistributed.cpp

```diff
@@ -742,12 +742,16 @@ void StorageDistributed::read(
             remote_storage_id,
             remote_table_function_ptr);
         header = InterpreterSelectQueryAnalyzer::getSampleBlock(query_tree_distributed, local_context, SelectQueryOptions(processed_stage).analyze());
+        /** For distributed tables we do not need constants in header, since we don't send them to remote servers.
```

Comment:
Are you sure we do not send constants to shards?

Reply:
We do not perform projection, but constants can be created in there

https://github.com/ClickHouse/ClickHouse/blob/3935c2bfb219817eab1b683a13eac1facec4aa09/src/Planner/Planner.cpp#L1513-L1518